### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by saverio-kantox**

### CODE REVIEW

File: kantox-obfuscator.gemspec [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+Gem::Specification.new do |s|
+  s.name        = 'kantox-obfuscator'
```
Comment (mparramon-kantox) [inline kantox-obfuscator.gemspec:4]: I would call it just `obfuscator`, it doesn't have anything Kantox-specific from what I can gather.

File: lib/kantox/obfuscator.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    option :keys, reader: :private, default: -> { [] }
+
+    def call(target)
+      case target
+      when String
+        obfuscate_string(target)
+      when Numeric
+        obfuscate_string(target.to_s)
+      when Hash
+        obfuscate_hash(target)
```
Comment (am-kantox) [inline lib/kantox/obfuscator.rb]:
```ruby
when Array then target.map(&method(:call))
```
It brings no harm, requires nothing special, but allows obfuscation of lists of items.

**Review by mparramon-kantox — APPROVED** [review]

File: spec/obfuscator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+require 'kantox-obfuscator'
+
+RSpec.describe Kantox::Obfuscator do
+  it 'obfuscates strings by default' do
+    expect(subject.call('1234567890')).to eq('12xxxx7890')
+  end
+
+  it 'obfuscates numbers by default' do
+    expect(subject.call(1_234_567_890)).to eq('12xxxx7890')
+    expect(subject.call(1_234_567_890.12)).to eq('12xxxxxxx0.12')
```
Comment (am-kantox) [inline spec/obfuscator_spec.rb:14]: Please do test what is less obvious, not what is as simple that literally might be tested inside your brain VM.
```ruby
expect(subject.call(1)).to WHAT?
expect(subject.call("1")).to WHAT?
expect(subject.call(nil)).to WHAT?
expect(subject.call({})).to WHAT?

```

**Review by am-kantox — CHANGES_REQUESTED** [review]

File: kantox-obfuscator.gemspec (hunk 1 above)
Comment (saverio-kantox) [reply kantox-obfuscator.gemspec:4]: There is already a gem called obfuscator, very old. I want to avoid name clashes

Comment (mparramon-kantox) [reply kantox-obfuscator.gemspec:4]: Maybe `obfuscaty` or something else? :)

Comment (am-kantox) [reply kantox-obfuscator.gemspec:4]: “enredo”

FWIW: my most helpful and hi-quality elixir packages are called “peluquero” and “tarearbol” respectively. And I know there is a tendency to call elixir packages “packagenamEX,” but I am not as dumb as to follow tendencies; we were born to build our own rules.

**Review by am-kantox — APPROVED** [review]

**Pull request merged**